TulparAudio/internal/include/tulpar/internal/SourceCollection.hpp

```diff
@@ -0,0 +1,333 @@
+/*
+* Copyright (C) 2018 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#ifndef TULPAR_INTERNAL_SOURCE_COLLECTION_HPP
+#define TULPAR_INTERNAL_SOURCE_COLLECTION_HPP
+
+#include <tulpar/internal/Collection.hpp>
+#include <tulpar/internal/Context.hpp>
+
+#include <tulpar/internal/BufferCollection.hpp>
+
+#include <tulpar/audio/Buffer.hpp>
+#include <tulpar/audio/Source.hpp>
+
+#include <tulpar/TulparAudio.hpp>
+
+#include <array>
+#include <unordered_map>
+#include <vector>
+
+namespace tulpar
+{
+namespace internal
+{
+
+/** @brief  Handles basic source id operations */
+struct OpenAVSourceHandler
+{
+    /** @brief  Generates a collection of source handles via OpenAL
+     *
+     *  @param  batchSize   number of handles to generate
+     *
+     *  @return a collection of newly generated source handles
+     */
+    static Collection<audio::Source>::Handles Generate(uint32_t batchSize);
+
+    /** @brief  Resets given handle
+     *
+     *  Stops given source
+     *
+     *  @param  handle  valid source handle
+     */
+    static void Reclaim(Collection<audio::Source>::Handle handle);
+
+    /** @brief  Returns given collection of handles to OpenAL
+     *
+     *  @param  handle  collection of valid source handles
+     */
+    static void Delete(Collection<audio::Source>::Handles const& handle);
+};
+
+/** @brief  Collection working with audio sources */
+class SourceCollection
+    : public Collection<audio::Source>
+{
+public:
+    //! Shortcut to source handle type
+    using SourceHandle = audio::Source::Handle;
+
+    //! Shortcut to buffer handle type
+    using BufferHandle = audio::Buffer::Handle;
```

Comment:
Maybe this namespace naming is not helping that much if it has to be covered with aliases? I would personally hate to look it up every time. Maybe it's better to rename namespaces so they would be shorter and more convenient to use?

Reply:
This file is a part of `internal` structure that's not published when the library is installed/used.

> I would personally hate to look it up every time.

Hate to look up what? It's the idea of `using` types not to have to know the final type. This aliasing is done to improve readability of the interface.

What's your specific suggestion? Is it to remove `audio` namespace completely?
The `tulpar::audio` namespace is there to distinguish library level namespace `tulpar` from underlying mechanisms (in the case for tulpar only `audio` is visible to the end user).